addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2982,150 +2982,121 @@ const SelectPagerUserValueWidget = SelectUserValueWidget.extend({
-
-        this.inputEl = document.createElement('input');
-        this.inputEl.setAttribute('placeholder', _t("Search for records..."));
-        const searchEl = document.createElement('div');
-        searchEl.classList.add('o_we_m2o_search');
-        searchEl.appendChild(this.inputEl);
-        this.menuEl.appendChild(searchEl);
-
-        this.searchMore = document.createElement('div');
-        this.searchMore.classList.add('o_we_m2o_search_more');
-        this.searchMore.textContent = _t("Search more...");
-        this.searchMore.title = _t("Search to show more records");
-
-        if (this.options.createMethod) {
-            this.createInput = new InputUserValueWidget(this, undefined, {
-                classes: ['o_we_large'],
-                dataAttributes: { noPreview: 'true' },
-            }, this.$target);
-            this.createButton = new ButtonUserValueWidget(this, undefined, {
-                classes: ['flex-grow-0'],
-                dataAttributes: {
-                    noPreview: 'true',
-                    [this.options.createMethod]: '', // Value through getValue.
-                },
-                childNodes: [document.createTextNode(_t("Create"))],
-            }, this.$target);
-            // Override isActive so it doesn't show up in toggler
-            this.createButton.isActive = () => false;
-
-            await Promise.all([
-                this.createInput.appendTo(document.createDocumentFragment()),
-                this.createButton.appendTo(document.createDocumentFragment()),
-            ]);
-            this.registerSubWidget(this.createInput);
-            this.registerSubWidget(this.createButton);
-            // TODO @owl-options: rename `createWidget` once m2o and m2m widgets are
-            // refactored, into something OWL-y (probably `createTemplate`?).
-            this.createWidget = _buildRowElement('', {
-                classes: ['o_we_full_row', 'o_we_m2o_create', 'p-1'],
-                childNodes: [this.createInput.el, this.createButton.el],
-            });
+        this.options.limit = parseInt(this.options.limit);
+        this.options.fields = JSON.parse(this.options.fields);
+        if (!this.options.fields.includes('display_name')) {
+            this.options.fields.push('display_name');
         }
+        this.options.domain = JSON.parse(this.options.domain);
+        this.options.domainComponents = {};
+        this.options.nullText = this.$target[0].dataset.nullText ||
+            JSON.parse(this.$target[0].dataset.oeContactOptions || '{}')['null_text'];
+    }
+    get records() {
+        return this._state.records;
+    }
+    set records(value) {
+        this._state.records = value;
+    }
+    get hasMore() {
+        return this._state.hasMore;
+    }
+    get waitingForSearch() {
+        return this._state.waitingForSearch;
+    }
+    set waitingForSearch(value) {
+        this._state.waitingForSearch = value;
+    }
```

Comment:
Why does this needs to be in this._state? It doesn't seem to be used in the template, just in an event handler. Do we need to re-render when it changes?

Reply:
It doesn't. It's even worse than that: that doesn't work at all because `waitingForSearch` and `afterSearch` are triggered before `state.records` has had time to be passed on.
Will be corrected in my next push.